### DISCUSSION TITLE
Custom rule

### DISCUSSION THREAD

**Discussion by fixytech** [dedoc/scramble#413] [Q&A] [answered]
hello good! I have a custom rule and I can't find a way to document it, I leave an example:

SetAvatarRequest.php

 public function rules(): array
    {

        return [

            'avatar' => [
                'required',
                'string',
                new Base64ImageSizeRule()
            ]
        ];
    }

![Captura desde 2024-06-07 23-40-21](https://github.com/dedoc/scramble/assets/50503561/2769c154-3e82-4723-9061-db22c11b3311)

Thank you so much!

**Comment by romalytvynenko** [comment, marked as answer]
@fixytech hey!

You can implement method `docs` on your rule class:

```php
use Dedoc\Scramble\Support\Generator\Types\Type as OpenApiType;

public function docs(OpenApiType $prevType, TypeTransformer $openApiTransformer): OpenApiType;
```

For example:
```php
use Dedoc\Scramble\Support\Generator\Types\Type as OpenApiType; 
use Dedoc\Scramble\Support\Generator\Types\StringType; 

public function docs(OpenApiType $prevType, TypeTransformer $openApiTransformer): OpenApiType
{
    return (new StringType)->format('password');
} 
```

This is not documented feature as in the near future I plan to change namespaces and potentially the API for this feature.

But for now it should do the work.
